content/src/service/access/AccessCheckerForWearables.ts

```diff
@@ -48,46 +48,57 @@ export class AccessCheckerForWearables {
 
     private async checkCollectionAccess(collection: string, itemId: number, ethAddress: EthAddress): Promise<boolean> {
         try {
-            const collectionItems: WerableCollectionItems = await this.getCollectionItems(collection, itemId, ethAddress)
-            return collectionItems.collections[0]?.creator === ethAddress
-                || collectionItems.collections[0]?.managers.includes(ethAddress)
-                || collectionItems.items[0]?.managers.includes(ethAddress)
+            const ethAddressLowercase = ethAddress.toLocaleLowerCase()
+            const permissions: WearableItemPermissionsData = await this.getCollectionItems(collection, itemId, ethAddressLowercase)
+            return permissions.collectionCreator === ethAddressLowercase
+                || permissions.collectionManagers.includes(ethAddressLowercase)
+                || permissions.itemManagers.includes(ethAddressLowercase)
         } catch (error) {
             this.LOGGER.error(`Error checking wearable access (${collection}, ${itemId}, ${ethAddress}).`, error)
             return false
         }
     }
 
-    private async getCollectionItems(collection: string, itemId: number, ethAddress: EthAddress): Promise<WerableCollectionItems> {
+    private async getCollectionItems(collection: string, itemId: number, ethAddress: EthAddress): Promise<WearableItemPermissionsData> {
         const query = `
          query getCollectionRoles($collection: String!, $itemId: Int!) {
             collections(where:{id: $collection}) {
-              minters
-              managers
               creator
-            }
-            items(where:{collection: $collection, itemId: $itemId}) {
-              itemId
+              managers
               minters
+            }
+            items(where:{collection: $collection, blockchainId: $itemId}) {
               managers
+              minters
             }
         }`
 
         try {
-            return await this.fetcher.queryGraph<WerableCollectionItems>(
+            const wearableCollectionsAndItems = await this.fetcher.queryGraph<WearableCollectionsAndItems>(
                 this.dclCollectionsAccessUrl,
                 query,
                 { collection: collection, itemId: itemId})
+            return {
+                collectionCreator : wearableCollectionsAndItems.collections[0]?.creator,
```

Comment:
Is it possible that `wearableCollectionsAndItems.collections` or `wearableCollectionsAndItems.items` is empty? If so, then we are not handing the `undefined` in `checkCollectionAccess`, and it would fail

Reply:
It's covered by the try/catch, but I'm including an extra check just to avoid the error log.